src/inspector/inspectorhandler.cpp

```diff
@@ -661,20 +511,66 @@ static QList<InspectorCommand> s_commands{
         [](InspectorHandler*, const QList<QByteArray>& arguments) {
           QJsonObject obj;
 
-          for (const InspectorSettingCommand& setting : s_settingCommands) {
-            if (arguments[1] == setting.m_settingName) {
-              obj["value"] = setting.m_get();
-              return obj;
-            }
+          SettingsHolder* settingsHolder = SettingsHolder::instance();
+          const QMetaObject* metaObject = settingsHolder->metaObject();
+          int propertyId = metaObject->indexOfProperty(arguments[1]);
+
+          if (propertyId < 0) {
+            obj["error"] =
+                QString("Property %1 is not exposed by SettingsHolder")
+                    .arg(QString(arguments[1]));
+            return obj;
           }
 
-          QStringList settings;
-          for (const InspectorSettingCommand& setting : s_settingCommands) {
-            settings.append(setting.m_settingName);
+          QMetaProperty property = metaObject->property(propertyId);
+          Q_ASSERT(property.isValid());
+
+          QJsonValue value;
+          switch (property.type()) {
+            case QVariant::Bool:
```

Comment:
QVariant can handle quite a few cases out of the box if we don't want to manually limit the types: 

```
QMetaProperty property = metaObject->property(propertyId);
QVariant prop_value = property.read(settingsHolder);
if (prop_value.canConvert<QJsonValue>()) {
  obj["value"] = prop_value.toJsonValue();
  return obj;
}
else if (prop_value.canConvert<QJsonArray>()){
   obj["value"] = prop_value.toJsonArray();
  return obj;
}
switch (property.type()) {
            case QVariant::DateTime: // <- Really not sure if this is already supported out of the box.
            .....
}
```

Reply:
oh! nice!!